Source/Core/DolphinQt2/Settings/GameCubePane.cpp

```diff
@@ -302,25 +314,28 @@ void GameCubePane::SaveSettings()
 
     switch (index)
     {
-    case 0:
+    case EXP_NOTHING:
       dev = ExpansionInterface::EXIDEVICE_NONE;
       break;
-    case 1:
+    case EXP_DUMMY:
       dev = ExpansionInterface::EXIDEVICE_DUMMY;
       break;
-    case 2:
-      dev = ExpansionInterface::EXIDEVICE_MEMORYCARD;
+    case EXP_MEMORYCARD:
+      if (i == 3)
```

Comment:
Could you please leave a comment why `i == 3` is special?

Reply:
I think it's clearer now, also the magic number was wrong anyway :/ (i < 3 loops should never reach 3 :smile:)